web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,18 +34,36 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
-  ).value;
+  const status = await connection.getSignatureStatus(signature);
```

Comment:
Rather than to check the signature status in a loop, did you consider making a _subscription_ using `onSignature(…)`? That way, the client can chill out and wait for the server to _push_ updates to the status, if any, while it keeps polling for the latest block height.

Reply:
We noticed that this was being done in the original `confirmTransaction` and we did try to implement the same logic here, but it got really messy with all of the imports we needed so we abandoned it for the time being. We can definitely explore it again.